### PR TITLE
New InputButton component

### PR DESCRIPTION
**Pull request by augustocb**
#### What is the purpose of this pull request?

Creates the InputButton component.

#### What problem is this solving?

It improves the experience of using a one-input-form.

Details in Figma: https://www.figma.com/file/2t4pmT0LUkPYbohaSxxJ7k/03-InputButton?node-id=0%3A1

#### How should this be manually tested?

Run `yarn styleguide` and go to http://localhost:6060/#/Components/Forms/InputButton.

#### Screenshots or example usage

Screen Shot 2019-10-11 at 19 46 10

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by ericreis** [comment]
I am not sure if this should be solved with a prop or a different component (**InputSubmit**) like other use cases (InputSearch,  InputCurrency, InputPassword, etc.)

**Comment by augustocb** [comment]
@ericreis yes, it should and it has been designed this way!

File: react/components/Input/index.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 
+import Button from '../Button'
```
Comment (emersonlaurentino) [inline react/components/Input/index.js]: I think you should import the button into the new component and pass it as `suffix` to Input

**Comment by emersonlaurentino** [comment]
I believe this component is not necessary as it is possible to create its behavior with the current input

```jsx
  <Input
    placeholder="Placeholder"
    suffixClasses="flex items-center mr1"
    small="small"
    suffix={
      <Button 
        size="small"
        type="submit"
        variation="secondary"
      >
        Submit
      </Button>
    }
  />
```

**Comment by cmdalbem** [comment]
I tend to agree with Emerson's point. We could add an usage example in the Input component showing how to accomplish this along with the appropriate usage guidelines for this scenario.

**Comment by omninando** [comment]
@emersonlaurentino @cmdalbem they are not possible due to wrong spacing between input and button.

Screen Shot 2019-10-23 at 11 14 03

**Comment by emersonlaurentino** [comment]
@omninando you can add a negative margin to right and it's settled. It's not the most beautiful way to solve it, but I think it's better than creating a new component just for that.

This is a problem that we might have anticipated in `suffix` creation since the "Calculate" button is an input suffix.

**Comment by kevinch** [comment]
Disagree with the negative margin hack. If it's the only solution then we should publish this new component.

**Comment by ericreis** [comment]
Why doesn't it follow the **InputSearch** behavior? Since it's the same idea (an input with a button) I think we should try our best to maintain the whole experience.

cc: @augustocb @cmdalbem 

image

**Comment by augustocb** [comment]
@ericreis don't worry! we are mapping and redesigning all components related to input with buttons.

**Comment by ericreis** [comment]
I reverted these changes since they were merged into master in an unusual way.

cc @augustocb (just to document why I reverted)

**Pull request merged**